Assets/Scripts/DynamicCamera/CameraBehavior.cs

```diff
@@ -0,0 +1,76 @@
+﻿namespace HarmonyQuest.DynamicCamera
+{
+    using UnityEngine;
+
+    public abstract class CameraBehavior : MonoBehaviour
```

Comment:
I believe it does. All it's children need update and awake functions and I don't know if it'd work as an interface.

Reply:
This could be made to be a regular C# classes. All the code that needs to be called on Update and Awake can be called by the CameraController on each CameraBehavior attached to it. I'd think we're better off doing that as it makes sure that the CameraController has control over all the CameraBehaviors. As this is, we could potentially end up with a situation where a CameraBehavior receives updates and moves the camera without the CameraController knowing at all. The horror!

Reply:
Got it working, the big drawback to this though is the is not being able to adjust the camera behavior fields from the inspector, but I can just write an editor script for that if I'm really missing it.